src/core/store/modules/lines/lines.reducer.ts

```diff
@@ -105,9 +109,20 @@ const linesReducer = createReducer(linesInitialState, (builder) => {
       state.selectedPosition = position;
     })
 
+    .addCase(setSelectedLinePositionProposal, (state, { payload: { position, proposal } }) => {
+      state.selectedProposal = proposal;
+    })
+
     .addCase(setPosition, (state, { payload: { id, position } }) => {
       state.positionsMap[id] = position;
     })
+
+    .addCase(setProposal, (state, { payload: { lineAddress, positionId, proposalId } }) => {
+      const { revokedAt: oldRevokedAt, ...rest } = state.positionsMap[positionId]?.proposalsMap[proposalId];
+      const revokedAt = 1234; // TODO: replace with actual time the proposal was revoked
+      const updatedProposal = { revokedAt, ...rest };
```

Comment:
`{...rest, revokedAt  }`, right now you are overwriting it with `oldRevokedAt` immediately after you set the new value

Reply:
thanks for the tip - fixed!